crates/anvil-polkadot/src/substrate_node/snapshot.rs

```diff
@@ -18,12 +20,17 @@ pub struct SnapshotManager {
     client: Arc<Client>,
     backend: Arc<Backend>,
     next_snapshot_id: U256,
-    snapshots: BTreeMap<U256, Snapshot>,
+    snapshots: Arc<Mutex<BTreeMap<U256, (Snapshot, Hash)>>>,
```

Comment:
I think we don't need `Arc<Mutex`. API requests are serialized, so we shouldn't worry about concurrent access here. Snapshot manager is also only accessed in RPCs.